src/modules/fancyzones/lib/WindowMoveHandler.cpp

```diff
@@ -47,7 +48,11 @@ class WindowMoveHandlerPrivate
 public:
     WindowMoveHandlerPrivate(const winrt::com_ptr<IFancyZonesSettings>& settings) :
         m_settings(settings)
-    {};
+    {
+        secondaryMouseButtonHook = new LLMouseHookWrapper((GetSystemMetrics(SM_SWAPBUTTON) ? WM_LBUTTONDOWN : WM_RBUTTONDOWN), std::bind(&WindowMoveHandlerPrivate::UpdateMouseSwitchState, this));

```

Comment:
we're leaking memory here, so we need to use `std::unique_ptr` or `std::optional`